numpy/core/tests/test_numeric.py

```diff
@@ -2748,3 +2748,7 @@ def test_zero_dimension(self):
         td = np.tensordot(a, b, (1, 0))
         assert_array_equal(td, np.dot(a, b))
         assert_array_equal(td, np.einsum('ij,jk', a, b))
+        # Test resolution to issue #12130
+        arr_0d = np.array(1)
+        ret = np.tensordot(arr_0d, arr_0d, ([], []))  # contracting no axes is well defined
```

Comment:
perhaps "# empty axes defaults to (2,)"

Reply:
What do you mean "defaults to (2,)"? Empty axis just means "do not contract anything". It works for higher dimensions too:
```
a = np.zeros((2, 3))
b = np.zeros((4, 5))
np.tensor_dot(a, b, ([], [])).shape  # (2, 3, 4, 5)
```